### PR TITLE
improve secret usage

### LINKED ISSUE #73: Feature Request: use existing secret
I wanna use a existing secret, so we can use this chart with encrypted secrets e.g. sealed secrets or sops.
Let me know if you like the idea, so I can pull request this feature.

### PR DESCRIPTION
**Pull request by grafjo**
#### What this PR does / why we need it:
make existing secrets usable - e.g. sealed secrets / mozilla sop

#### Which issue this PR fixes
  - fixes #73 

#### Checklist
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

**Comment by grafjo** [comment]
@monotek what do you think? I know - needs some polishing to be mergeable, the base idea is more important part

**Review by monotek — CHANGES_REQUESTED** [review]
Please raise chart version so the install step works.
I will review when all tests are grean.

**Comment by grafjo** [comment]
Ok tests are green

**Comment by grafjo** [comment]
@monotek please let me know if you like the change, so I'll add some documentation into README.md

**Comment by monotek** [comment]
Sorry, i'm realy busy at the moment...
Will do it as soon as possible.

File: zammad/templates/statefulset.yaml [hunk 1]
```diff
@@ -74,6 +74,13 @@ spec:
         - name: "{{ $key }}"
           value: "{{ $value }}"
         {{- end }}
+        {{ if or .Values.envConfig.elasticsearch.pass .Values.secrets.elasticsearch.useExisting }}
+        - name: "ELASTICSEARCH_PASSWORD"
```
Comment (monotek) [inline zammad/templates/statefulset.yaml:78]: why is this env var added?
imho its not used in the container.

**Review by monotek — CHANGES_REQUESTED** [review]

File: zammad/templates/statefulset.yaml (hunk 1 above)
Comment (grafjo) [reply zammad/templates/statefulset.yaml:78]: it's used in the elasticsearch-init script, when username / passwort is required to talk with elasticsearch during elasticsearch initialisation

Comment (monotek) [reply zammad/templates/statefulset.yaml:78]: Which step you're talking about exactly?
The credentials are directly used from the variables: https://github.com/zammad/zammad-helm/blob/master/zammad/templates/configmap-init.yaml#L14-L15

File: zammad/templates/configmap-init.yaml [hunk 2]
```diff
@@ -10,10 +10,11 @@ data:
     #!/bin/bash
     set -e
     bundle exec rails r 'Setting.set("es_url", "{{ .Values.envConfig.elasticsearch.schema }}://{{ if .Values.elasticsearch.enabled }}zammad-master{{ else }}{{ .Values.envConfig.elasticsearch.host }}{{ end }}:{{ .Values.envConfig.elasticsearch.port }}")'
-    {{- if and .Values.envConfig.elasticsearch.user .Values.envConfig.elasticsearch.pass }}
-    bundle exec rails r 'Setting.set("es_user", "{{ .Values.envConfig.elasticsearch.user }}")'
-    bundle exec rails r 'Setting.set("es_password", "{{ .Values.envConfig.elasticsearch.pass }}")'
-    {{ end }}
+    ELASTICSEARCH_USER=${ELASTICSEARCH_USER:-{{ .Values.envConfig.elasticsearch.user }}}
+    if [ -n "${ELASTICSEARCH_USER}" ] && [ -n "${ELASTICSEARCH_PASSWORD}" ]; then
```
Comment (grafjo) [inline zammad/templates/configmap-init.yaml:14]: @monotek have a look here

Comment (monotek) [reply zammad/templates/configmap-init.yaml:14]: ah... now is see. it's used below.

**Review by monotek — APPROVED** [review]

**Comment by grafjo** [comment]
@monotek thx for merging - maybe you overlooked the `Variables are documented in the README.md` part as a todo from my side. I'll open an other PR to fix that!

**Pull request merged**